emperor/flat_plots.py

```diff
@@ -0,0 +1,346 @@
+from matplotlib import use
+use('Agg')  # noqa
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+
+def plot_3x3(ordination, offset, mapping, field, colors, null_kw=None,
```

Comment:
I have to acknowledge that I did not realize from your original demonstration (during the EMP meeting), that the resulting plots actually display up to PC6. I misunderstood this, my apologies for overlooking that. While I understand your concerns with managing space distribution (and all the cumbersome matplotlib juggling that that entails), it seems like there could still be a way to guarantee consistency, if we opt for a slightly different layout. In the image below I have drawn two cases, when someone selects PC1, PC2 & PC3 and when someone selects PC1, PC2, PC3 & PC4.

(tilt your head 🙃 - there's no 90º tilted emoji )
![img_4502](https://cloud.githubusercontent.com/assets/375307/20360438/9dce1d72-abe8-11e6-9abd-202878191788.JPG)

What would essentially be getting drawn are the upper diagonals of the matrix of combinations of the categories. And this would open up space for a different placement of the histogram. It is true that this functionality differs a lot from what is currently implemented, so do let me know if this is not something you would be interested in implementing. Regardless, I think that's the API we want to have, providing only 3x3 plots seems rather arbitrary and inflexible.

Adding shapes based on a category will be very useful, but we should defer that to a separate iteration.